pyro/infer/mcmc/util.py

```diff
@@ -0,0 +1,137 @@
+import torch
+from collections import defaultdict
+
+from pyro.distributions.util import log_sum_exp
```

Comment:
Could we try to use the `logsumexp` form internally, since that's what PyTorch 0.4.1 calls it?

Reply:
Oh..sorry, I forgot to update. Will do so.